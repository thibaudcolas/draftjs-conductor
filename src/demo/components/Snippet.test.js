import React from "react";
import { shallow } from "enzyme";
import { convertFromRaw } from "draft-js";

import Snippet from "./Snippet";

describe("Snippet", () => {
  it("renders", () => {
    const content = convertFromRaw({
      entityMap: {
        0: {
          type: "SNIPPET",
          data: {
            text: "This is a snippet",
          },
        },
      },
      blocks: [
        {
          key: "a",
          text: " ",
          entityRanges: [
            {
              offset: 0,
              length: 1,
              key: 0,
            },
          ],
        },
      ],
    });

    expect(
      shallow(
        <Snippet contentState={content} block={content.getFirstBlock()} />,
      ),
    ).toMatchSnapshot();
  });

  it("no entity", () => {
    const content = convertFromRaw({
      entityMap: {},
      blocks: [
        {
          key: "a",
          text: " ",
        },
      ],
    });

    expect(
      shallow(
        <Snippet contentState={content} block={content.getFirstBlock()} />,
      ),
    ).toMatchSnapshot();
  });
});
