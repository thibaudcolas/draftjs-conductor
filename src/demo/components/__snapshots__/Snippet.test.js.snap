// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snippet no entity 1`] = `
<div
  className="Snippet"
  contentEditable={false}
>
  This is a snippet block:
  <div
    className="Snippet__text"
    contentEditable={false}
  >
    Placeholder
  </div>
</div>
`;

exports[`Snippet renders 1`] = `
<div
  className="Snippet"
  contentEditable={false}
>
  This is a snippet block:
  <div
    className="Snippet__text"
    contentEditable={false}
  >
    This is a snippet
  </div>
</div>
`;
